src/foam/nanos/crunch/ui/CapabilityWizardlet.js

```diff
@@ -50,8 +51,24 @@ foam.CLASS({
       name: 'title',
       class: 'String',
       expression: function(capability) {
-        if ( ! capability || ! capability.name ) return '';
-        return capability.name;
+        var expr = foam.mlang.Expressions.create();
+        return this.localeDAO.where(
+          expr.AND(
+            expr.OR(
+              expr.EQ(foam.i18n.Locale.LOCALE, foam.locale),
+              expr.EQ(foam.i18n.Locale.LOCALE, foam.locale.substring(0,foam.locale.indexOf('-')))),
+            expr.EQ(foam.i18n.Locale.ID, capability.id + '.name')))
+        .select().then(function(a){
+          let arr = a.array;
+          if ( arr.length > 0 ) {
+            let ea = arr[0];
+            return ea.target;
+          }
+          return capability.name;
```

Comment:
since this is in a then function - I think it's safer to swap 'return capability.name' to 'this.title = capability.name'

Reply:
This is technically broken either way. Just returning the value doesn't work because expressions need to return the value directly, not a promise, which is what happens when you have an async DAO call. If you do what Anna says, it will work better, but it won't be an expression anymore because if capability changes again then it won't regenerate since you've explicitly set the value, which probably isn't an issue in this case because you're probably only setting the capability once, in which case, you could just set the postSet on Capability to set the title.

Reply:
Or better, just use a factory.

Reply:
noted and fixed

Reply:
> Or better, just use a factory.

for some reason it doesn't work